app/scripts/controllers/home.ctrl.js

```diff
@@ -1,19 +1,42 @@
 'use strict';
 
-angular.module('yourStlCourts').controller('HomeCtrl', function (TicketFinder,Municipalities,$state) {
+angular.module('yourStlCourts').controller('HomeCtrl', function (TicketFinder, $state, municipalities, courts) {
   var ctrl = this;
-  ctrl.municipalities = Municipalities.municipalitiesMapNames();
-  ctrl.selectedMunicipality = null;
+  ctrl.citySearchGroups = [];
+  ctrl.municipalities = municipalities;
+  ctrl.selectedCitySearchGroup = null;
   ctrl.finderSelected = TicketFinder.TicketFinderToSelect.NONE;
-  ctrl.updateFinderSelected = function(TicketFinderToSelect){
-    ctrl.finderSelected = TicketFinderToSelect;
+
+  ctrl.updateFinderSelected = function(ticketFinderToSelect){
+    ctrl.finderSelected = ticketFinderToSelect;
   };
 
   ctrl.clearTicketFinder = function(){
     ctrl.finderSelected = TicketFinder.TicketFinderToSelect.NONE;
   };
 
-  ctrl.municipalitySelected  = function(){
-    $state.go('courtSearchInfo',{courtId : ctrl.selectedMunicipality.court_id});
+  ctrl.onCitySearchGroupSelected  = function(){
+    $state.go('courtSearchInfo', {courtId : ctrl.selectedCitySearchGroup.court.id});
+  };
+
+  ctrl.groupCourts = function(citySearchGroup) {
+    return citySearchGroup.municipalityCourtCount > 1 ? citySearchGroup.municipalityName : undefined; //undefined causes no group to be made
   };
+
+  function populateCitySearchGroups() {
+    var groups = [];
+    municipalities.forEach(function(municipality) {
+      municipality.courts.forEach(function(courtId) {
+        groups.push({
+          municipalityName: municipality.name,
+          municipalityCourtCount: municipality.courts.length,
+          court: _.find(courts, {id: parseInt(courtId)})
```

Comment:
Everything with "_." is from the lodash library. It's a bunch of utility methods for arrays and objects, pretty much.